### PR TITLE
Vaccination metrics

### LINKED ISSUE #233: Get the daily and total amount of first and second vaccination doses
1. Create the new collection [`vaccinations`](https://github.com/covid19cz/erouska-backend/wiki/Database#vaccinations)
2. Create the new function [`downloadAndCountVaccinations`](https://github.com/covid19cz/erouska-backend/wiki/Firebase-Functions#downloadandcountvaccinations)

### PR DESCRIPTION
**Pull request by dizider**
Closes #233

### CODE REVIEW

File: internal/functions/coviddata/download-and-count-vaccinations.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	logger.Debugf("Fetched new vaccinations metrics: %+v", lastDateVaccinations)
+
+	return &lastDateVaccinations, nil
+}
+
+// DownloadAndCountVaccinations downloads vaccinations metrics json and writes it to firestore
+func DownloadAndCountVaccinations(w http.ResponseWriter, r *http.Request) {
+	var ctx = context.Background()
+	logger := logging.FromContext(ctx)
+	client := store.Client{}
```
Comment (jendakol) [inline internal/functions/coviddata/download-and-count-vaccinations.go]: Please
* Move this function to the top of the file
* Use named loggers :-)
* Use this public function only to create a configuration that is later passed to another function that does the job (as is in most of the other functions, look at `EfgsUploadKeys` for example

**Review by jendakol — CHANGES_REQUESTED** [review]

File: internal/functions/coviddata/download-and-count-vaccinations.go [hunk 2]
```diff
@@ -13,10 +13,34 @@ import (
...
+		return
+	}
+
+	httputils.SendResponse(w, r, struct{ status string }{status: "OK"})
+}
+
+func fetchVaccinationsData(client HTTPClient) (*VaccinationsAggregatedData, error) {
 	var ctx = context.Background()
-	logger := logging.FromContext(ctx)
+	logger := logging.FromContext(ctx).Named("FetchVaccinationData")
```
Comment (jendakol) [inline internal/functions/coviddata/download-and-count-vaccinations.go]: Casing difference.

File: internal/functions/coviddata/download-and-count-vaccinations.go [hunk 3]
```diff
@@ -87,32 +111,18 @@ func fetchVaccinationsData(client HTTPClient) (*VaccinationsAggregatedData, erro
...
 }
 
-// DownloadAndCountVaccinations downloads vaccinations metrics json and writes it to firestore
-func DownloadAndCountVaccinations(w http.ResponseWriter, r *http.Request) {
+func persistVaccinationsData(data *VaccinationsAggregatedData) error {
+
 	var ctx = context.Background()
-	logger := logging.FromContext(ctx)
+	logger := logging.FromContext(ctx).Named("PersistVaccinationsData")
 	client := store.Client{}
```
Comment (jendakol) [inline internal/functions/coviddata/download-and-count-vaccinations.go]: This should be given as an argument.

File: internal/functions/coviddata/download-and-count-vaccinations.go [hunk 4]
```diff
@@ -13,10 +13,34 @@ import (
 	"time"
 )
 
-func fetchVaccinationsData(client HTTPClient) (*VaccinationsAggregatedData, error) {
+// DownloadAndCountVaccinations downloads vaccinations metrics json and writes it to firestore
+func DownloadAndCountVaccinations(w http.ResponseWriter, r *http.Request) {
+	var ctx = context.Background()
+	logger := logging.FromContext(ctx).Named("DownloadAndCountVaccination")
+
+	spaceClient := http.Client{
```
Comment (jendakol) [inline internal/functions/coviddata/download-and-count-vaccinations.go]: WTH is "space"? :smile:

**Review by jendakol — CHANGES_REQUESTED** [review]

File: internal/functions/coviddata/download-and-count-vaccinations.go (hunk 4 above)
Comment (dizider) [reply internal/functions/coviddata/download-and-count-vaccinations.go]: This naming is used in `DownloadCovidDataTotal` ..... 🤷

Comment (jendakol) [reply internal/functions/coviddata/download-and-count-vaccinations.go]: OK, maybe fix it there too :joy: Wasn't it you who wrote it there?

File: internal/functions/coviddata/download-and-count-vaccinations.go (hunk 3 above)
Comment (jendakol) [reply internal/functions/coviddata/download-and-count-vaccinations.go]: It's still here :man_facepalming:

**Review by jendakol — CHANGES_REQUESTED** [review]

**Review by jendakol — APPROVED** [review]

**Pull request merged**